src/langchain_google_bigtable/chat_message_history.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Bigtable-based chat message history"""
+from __future__ import annotations
+
+import json
+import re
+import time
+import uuid
+from typing import List, Optional
+
+from google.cloud import bigtable
+from google.cloud.bigtable.row_filters import RowKeyRegexFilter
+from langchain_core.chat_history import BaseChatMessageHistory
+from langchain_core.messages import BaseMessage, messages_from_dict
+
+COLUMN_FAMILY = "langchain"
+COLUMN_NAME = "history"
+
+default_client: Optional[bigtable.Client] = None
+
+
+class BigtableChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history that stores history in Bigtable.
+
+    Args:
+        instance_id: The Bigtable instance to use for chat message history.
+        table_id: The Bigtable table to use for chat message history.
+        session_id: The session ID.
+        client : Optional. The pre-created client to query bigtable.
+    """
+
+    def __init__(
+        self,
+        instance_id: str,
+        table_id: str,
+        session_id: str,
+        client: Optional[bigtable.Client] = None,
+    ) -> None:
+        self.client = (
+            (client or self.__get_default_client())
+            .instance(instance_id)
+            .table(table_id)
+        )
+
+        self.session_id = session_id
+        self.__init_schema()
+
+    def __get_default_client(self) -> bigtable.Client:
+        global default_client
+        if default_client is None:
+            default_client = bigtable.Client(admin=True)
+        return default_client
```

Comment:
nit: I don't think this should be attached to the class -- I think we want to use it in all classes

Reply:
Will be done after this PR is merged